frontend/packages/dev-console/src/components/pipelines/modals/common/PipelineSecretSection.tsx

```diff
@@ -1,66 +1,50 @@
 import * as React from 'react';
-import { Formik, useField, useFormikContext, FormikValues } from 'formik';
+import { useFormikContext, FormikValues } from 'formik';
 import { PlusCircleIcon } from '@patternfly/react-icons';
 import { Button } from '@patternfly/react-core';
 import { ExpandCollapse } from '@console/internal/components/utils';
-import { SecretType } from '@console/internal/components/secrets/create-secret';
 import { SecretModel } from '@console/internal/models';
-import { k8sCreate } from '@console/internal/module/k8s';
-import {
-  associateServiceAccountToSecret,
-  getSecretAnnotations,
-} from '../../../../utils/pipeline-utils';
-import { SecretAnnotationId } from '../../const';
-import { advancedSectionValidationSchema } from './validation-utils';
+import { SecretKind } from '@console/internal/module/k8s';
+import { getSecretAnnotations } from '../../../../utils/pipeline-utils';
 import SecretForm from './SecretForm';
 import SecretsList from './SecretsList';
+import { secretFormDefaultValues } from './utils';
 
 import './PipelineSecretSection.scss';
 
-const initialValues = {
-  secretName: '',
-  annotations: { key: SecretAnnotationId.Image, value: '' },
-  type: SecretType.dockerconfigjson,
-  formData: {},
-};
-
 type PipelineSecretSectionProps = {
   namespace: string;
 };
 
 const PipelineSecretSection: React.FC<PipelineSecretSectionProps> = ({ namespace }) => {
-  const [secretOpenField] = useField<boolean>('secretOpen');
-  const { setFieldValue } = useFormikContext<FormikValues>();
+  const {
+    setFieldValue,
+    values: { newSecrets, secretForm },
+    setFieldTouched,
+  } = useFormikContext<FormikValues>();
 
-  const handleSubmit = (values, actions) => {
-    actions.setSubmitting(true);
+  const handleSubmit = () => {
+    setFieldValue(`secretForm`, secretFormDefaultValues);
+    setFieldTouched(`secretForm`, false);
     const newSecret = {
       apiVersion: SecretModel.apiVersion,
       kind: SecretModel.kind,
       metadata: {
-        name: values.secretName,
+        name: secretForm.secretName,
         namespace,
-        annotations: getSecretAnnotations(values.annotations),
+        annotations: getSecretAnnotations(secretForm.annotations),
       },
-      type: values.type,
-      stringData: values.formData,
+      type: secretForm.type,
+      stringData: secretForm.formData,
     };
-    k8sCreate(SecretModel, newSecret)
-      .then((resp) => {
-        actions.setSubmitting(false);
-        setFieldValue(secretOpenField.name, false);
-        associateServiceAccountToSecret(resp, namespace);
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        setFieldValue(secretOpenField.name, false);
-        actions.setStatus({ submitError: err.message });
-      });
+    const secrets: SecretKind[] = newSecrets;
+    secrets.push(newSecret as SecretKind);
```

Comment:
Avoid `.push` on arrays that are not created by you locally. It mutates the array and thus can have unintended side-effects like not updating the UI or updating at a later time if you do not submit back into the form.